### PR TITLE
SCRUM-2679 Enable loading of non-canonical biological entities

### PR DESCRIPTION
**Pull request by markquintontulloch**

### CODE REVIEW

File: src/main/java/org/alliancegenome/curation_api/jobs/executors/AgmDiseaseAnnotationExecutor.java [hunk 1]
```diff
@@ -75,11 +77,26 @@ public void runLoad(BulkLoadFile bulkLoadFile) {
...
+	public APIResponse runLoad(String dataProvider, List<AGMDiseaseAnnotationDTO> annotations) {
+
+		List<Long> annotationIdsLoaded = new ArrayList<>();
+		List<Long> annotationIdsBefore = new ArrayList<>();
+		annotationIdsBefore.addAll(agmDiseaseAnnotationDAO.findAllAnnotationIdsByDataProvider(dataProvider));
+		annotationIdsBefore.removeIf(Objects::isNull);
+		
+		BulkLoadFileHistory history = new BulkLoadFileHistory(annotations.size());
+		runLoad(history, dataProvider, annotations, annotationIdsLoaded);
+		runCleanup(diseaseAnnotationService, history, dataProvider, annotationIdsBefore, annotationIdsLoaded);
```
Comment (oblodgett) [inline src/main/java/org/alliancegenome/curation_api/jobs/executors/AgmDiseaseAnnotationExecutor.java]: @markquintontulloch the only draw back to this is if someone calls the API with a small set of Annotations then clean runs and deletes everything else. This was why I didn't have this runLoad do a cleanup.

Comment (oblodgett) [reply src/main/java/org/alliancegenome/curation_api/jobs/executors/AgmDiseaseAnnotationExecutor.java]: I kinda made the assumption that the API was only for bulk updates and not "loading"

Comment (markquintontulloch) [reply src/main/java/org/alliancegenome/curation_api/jobs/executors/AgmDiseaseAnnotationExecutor.java]: yeah, I was in two minds about this - maybe one for discussion with the group, but this is how it was previously

Comment (oblodgett) [reply src/main/java/org/alliancegenome/curation_api/jobs/executors/AgmDiseaseAnnotationExecutor.java]: And again these API were only suppose to serve the Integration tests, but I think that @cmpich ZFIN might be using them to do updates. And before they did not run the cleanup.

Comment (oblodgett) [reply src/main/java/org/alliancegenome/curation_api/jobs/executors/AgmDiseaseAnnotationExecutor.java]: `runLoad(history, speciesNames, alleles, null, null);` when the list is null no cleanup happens.

Comment (markquintontulloch) [reply src/main/java/org/alliancegenome/curation_api/jobs/executors/AgmDiseaseAnnotationExecutor.java]: I thought they did do cleanups (before you refactored the cleanup stuff) - that's certainly how they were working for the integration tests before those got refactored

Comment (oblodgett) [reply src/main/java/org/alliancegenome/curation_api/jobs/executors/AgmDiseaseAnnotationExecutor.java]: Ya... I guess when the cleanup code went in it did it on both.

**Comment by markquintontulloch** [comment]
Just realised I need to add more changes to this to sort out disease annotation curies

**Review by oblodgett — APPROVED** [review]

File: src/main/java/org/alliancegenome/curation_api/jobs/executors/AgmDiseaseAnnotationExecutor.java (hunk 1 above)
Comment (markquintontulloch) [reply src/main/java/org/alliancegenome/curation_api/jobs/executors/AgmDiseaseAnnotationExecutor.java]: @oblodgett - cleanup now removed for direct API calls

**Pull request merged**